src/preferences/PreferencesBase.js

```diff
@@ -1324,6 +1328,15 @@ define(function (require, exports, module) {
         },
 
         /**
+         * Returns a clone of all preferences defined.
+         *
+         * @return {Object}
+         */
+        getAllPreferences: function () {
+            return $.extend(true, {}, this._knownPrefs);
```

Comment:
Instead of `$.extend` for cloning you can use `_.clone` and `_.cloneDeep` from lowdash which is already included in the core.